src/lib.rs

```
@@ -1230,48 +1230,99 @@ impl EGraph {
         let mut contents = String::new();
         f.read_to_string(&mut contents).unwrap();
 
-        let span: Span = span!();
-        let mut actions: Vec<Action> = vec![];
-        let mut str_buf: Vec<&str> = vec![];
+        let mut parsed_contents: Vec<Vec<Value>> = Vec::with_capacity(contents.lines().count());
+
+        let mut row_schema = func.schema.input.clone();
+        if function_type.subtype == FunctionSubtype::Custom {
+            row_schema.push(func.schema.output.clone());
+        }
+
+        log::debug!("{:?}", row_schema);
+
+        let unit_val = self.backend.primitives().get(());
+
         for line in contents.lines() {
-            str_buf.clear();
-            str_buf.extend(line.split('\t').map(|s| s.trim()));
-            if str_buf.is_empty() {
+            let mut row: Vec<Value> = Vec::with_capacity(row_schema.len());
+
+            for (sort, raw) in row_schema.iter().zip(line.split('\t').map(|s| s.trim())) {
```

This will silently fail if a line is too long. You should collect the `line.split` and make sure it has the right length _before_ calling `zip`.